### PR TITLE
Fix #130 - Validate ip format in session

### LINKED ISSUE #130: Validate IP format in Session
Using an IPField.

### PR DESCRIPTION
**Pull request by flyusfly**

### CODE REVIEW

**Comment by flyusfly** [comment]
Let's discuss about it?

**Comment by flyusfly** [comment]
Let's discuss about it?

File: ban/db/fields.py [hunk 1]
```diff
@@ -199,6 +201,21 @@ def coerce(self, value):
         return value
 
 
+class IPAddressField(CharField):
+
+    max_length = 100
```
Comment (vinyll) [inline ban/db/fields.py]: I wonder if we should keep some generic 100 or try to make it 39 or 45 (http://stackoverflow.com/a/1076755/328117)

File: ban/db/fields.py [hunk 2]
```diff
@@ -199,6 +201,21 @@ def coerce(self, value):
...
+
+    def coerce(self, value=None):
```
Comment (vinyll) [inline ban/db/fields.py]: cf https://github.com/BaseAdresseNationale/ban/pull/163#discussion_r86297686

File: ban/tests/auth/test_models.py [hunk 3]
```diff
@@ -12,30 +10,3 @@ def test_user_password_is_hashed():
     user = models.User.get(models.User.id == user.id)
     assert user.password != password
```
Comment (vinyll) [inline ban/tests/auth/test_models.py]: No big deal here but I would also comfort checking the length of the string (32? depending on the hash)

File: ban/tests/auth/test_session.py [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+        'client': client.name
+    }
+
+
+def test_session_should_have_either_a_client_or_a_user():
+    with pytest.raises(ValueError):
+        models.Session.create()
+
+
+def test_session_can_be_created_with_a_client_and_an_valid_IPAddress():
```
Comment (vinyll) [inline ban/tests/auth/test_session.py]: lower case 'IPAddress'

File: ban/tests/auth/test_session.py [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+    client = ClientFactory()
+    session = models.Session.create(client=client, ip='192.168.0.1')
+    assert session.ip == '192.168.0.1'
+
+
+def test_session_can_not_be_created_with_a_client_and_an_invalid_IPAddress():
```
Comment (vinyll) [inline ban/tests/auth/test_session.py]: lower case

File: ban/tests/auth/test_session.py [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+    client = ClientFactory()
+    with pytest.raises(ValueError):
+        models.Session.create(client=client, ip='999.999.999.999')
+
+
+def test_session_can_be_created_with_a_user_and_an_valid_IPAddress():
```
Comment (vinyll) [inline ban/tests/auth/test_session.py]: lowercase

File: ban/tests/auth/test_session.py [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+    user = UserFactory()
+    session = models.Session.create(user=user, ip='192.168.0.1')
+    assert session.ip == '192.168.0.1'
+
+
+def test_session_can_not_be_created_with_a_user_and_an_invalid_IPAddress():
```
Comment (vinyll) [inline ban/tests/auth/test_session.py]: lowercase

File: ban/tests/auth/test_session.py [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+    user = UserFactory()
+    with pytest.raises(ValueError):
+        models.Session.create(user=user, ip='999.999.999.999')
```
Comment (vinyll) [inline ban/tests/auth/test_session.py]: Should we test with an IP v6?

**Pull request closed without merging**